contents/performance/Performance.md

```diff
@@ -0,0 +1,104 @@
+The 2 techniques you should know are:
+
+* Using appropriate Data Structures and Algorithms
+* Using Cache Friendly codes
+
+## Using Appropriate Data Structures and Algorithms
+
+Using appropriate Data Structures and Algorithms can improve the execution speed of your program trememdously. However, it requires time to learn and understand the nuances of each Data Structure and Algorithm and when to use them.
+
+A good example of how Data Structures and Algorithms can improve execution time is with [binary search](https://www.tutorialspoint.com/data_structures_algorithms/binary_search_algorithm.htm). When finding a particular element in a list, you need to search through the entire list. If it has a million entries, every find will require you to look through **1,000,000** entries. However, if you have a [sorted](https://www.tutorialspoint.com/data_structures_algorithms/sorting_algorithms.htm) list, you can use binary search to reduce the number of entries you have to look through to around **20**!
+
+If you are interested in learning more about Data Structures and Algorithms, you can learn more from the following resources:
+
+* [Geeksforgeeks.org](http://www.geeksforgeeks.org/data-structures/)
+* [Introduction to Algorithms (Book)](https://www.amazon.com/Introduction-Algorithms-3rd-MIT-Press/dp/0262033844)
+* [MIT online course](http://courses.csail.mit.edu/6.851/spring12/lectures/)
+* [topcoder.com](https://www.topcoder.com/community/data-science/data-science-tutorials/)
+
+There are more of such resources online. Just Google and you shall find.
+
+## Using Cache Friendly code
+
+Before we can talk about this, we need to know what the computer memory is. Computer memory has different components, [registers](https://en.wikipedia.org/wiki/Processor_register), [L1/L2/L3 cache](https://www.cs.umd.edu/class/fall2001/cmsc411/proj01/cache/cache.html), [RAM](https://en.wikipedia.org/wiki/Random-access_memory), and [disk](https://en.wikipedia.org/wiki/Hard_disk_drive) in order of their speed.
+
+> You can take a look at the extent of the difference of their access speeds [here](https://gist.github.com/jboner/2841832) <br>
+> You can also check out this [infographic](http://imgur.com/8LIwV4C)
+
+To understand this topic, you only need to know how the cache works. It is okay if you do not understand exactly how the other components, registers, RAM and disk work.
+
+The faster a cache is, the more expensive it is. This means that the L1 cache (a few kilobytes big) is smaller than the L2 cache (a few megabytes big) and so on, until the RAM (a few gigabytes big).
+
+Whenever data is requested, the computer will first look in the cache for the data. If it exists, this is known as a **cache hit**. If it does not exist, this is known as a **cache miss**. When a **cache miss** happens, a contiguous block of memory (not just the requested data) is retrieved and stored onto the cache. Because of this, we want to remember these rules to make cache friendly codes:
```

Comment:
I don't think this is really an explanation of cache that is suitable to talk about before the previous line. The reason why I did not explain what a cache is properly is because I kind of expect the user to use the links I provided to learn what a cache is first.

Do you think I should just include an explanation anyway? Cause it is kind of hard to summarise an explanation of cache in a few lines